### PR TITLE
[FIRRTL] Replace Lower Types with new version.

### PR DESCRIPTION
**Pull request by darthscsi**
Rename new version and remove old version.

### CODE REVIEW

File: lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp [hunk 1]
```diff
@@ -1,12 +1,29 @@
-//===- LowerTypes.cpp - Lower FIRRTL types to ground types ------*- C++ -*-===//
+//===- LowerBundleVectorTypes.cpp - Lower Aggregate Types -------*- C++ -*-===//
```
Comment (seldridge) [inline lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp:1]: Stale header.

File: lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
 //
 // Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
 // See https://llvm.org/LICENSE.txt for license information.
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
 //
-// This file implements the lowering of FIRRTL types to ground types.
+// This file defines the LowerBundleVectorTypes pass.  This pass replaces
```
Comment (seldridge) [inline lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp:9]:
```suggestion
// This file defines the LowerTypes pass.  This pass replaces
```

**Review by seldridge — APPROVED** [review]
🎊 LowerTypes is dead. Long live LowerTypes!

**Comment by fabianschuiki** [comment]
The net loss of lines of code makes me very happy :chart_with_downwards_trend:

**Comment by lattner** [comment]
woo hoo!

**Pull request merged**